lib/commands.py

```diff
@@ -111,7 +105,7 @@ def _ssh(hostname_or_ip, cmd, check=True, simple_output=True, suppress_fingerpri
     for line in iter(process.stdout.readline, b''):
         readable_line = line.decode(errors='replace').strip()
         stdout.append(line)
-        OUPUT_LOGGER.debug(readable_line)
+        logging.debug("> %s", readable_line)
```

Comment:
does this keep the realtime logging when there's a wait in progress etc?

Reply:
I see no reason why there would be a difference, are you thinking about anything specific?

Reply:
I used a specific output logger to have the log streamed to the user and without the prefix formated to differentiate the call and the answer:
```
Jul 25 10:49:48 INFO Wait for VM running
Jul 25 10:49:48 DEBUG [10.1.0.74] xe vm-param-get uuid=7901c950-2345-bcba-a524-796e9fcbf968 param-name=power-state
running
Jul 25 10:49:48 INFO Wait for VM IP
Jul 25 10:49:48 DEBUG [10.1.0.74] xe vm-param-get uuid=7901c950-2345-bcba-a524-796e9fcbf968 param-name=networks param-key=0/ip
Error: Key 0/ip not found in map
Jul 25 10:49:51 DEBUG [10.1.0.74] xe vm-param-get uuid=7901c950-2345-bcba-a524-796e9fcbf968 param-name=networks param-key=0/ip
Error: Key 0/ip not found in map
Jul 25 10:49:54 DEBUG [10.1.0.74] xe vm-param-get uuid=7901c950-2345-bcba-a524-796e9fcbf968 param-name=networks param-key=0/ip
Error: Key 0/ip not found in map
Jul 25 10:49:56 DEBUG [10.1.0.74] xe vm-param-get uuid=7901c950-2345-bcba-a524-796e9fcbf968 param-name=networks param-key=0/ip
Error: Key 0/ip not found in map
Jul 25 10:49:59 DEBUG [10.1.0.74] xe vm-param-get uuid=7901c950-2345-bcba-a524-796e9fcbf968 param-name=networks param-key=0/ip
Error: Key 0/ip not found in map
Jul 25 10:50:02 DEBUG [10.1.0.74] xe vm-param-get uuid=7901c950-2345-bcba-a524-796e9fcbf968 param-name=networks param-key=0/ip
Error: Key 0/ip not found in map
Jul 25 10:50:04 DEBUG [10.1.0.74] xe vm-param-get uuid=7901c950-2345-bcba-a524-796e9fcbf968 param-name=networks param-key=0/ip
Error: Key 0/ip not found in map
Jul 25 10:50:07 DEBUG [10.1.0.74] xe vm-param-get uuid=7901c950-2345-bcba-a524-796e9fcbf968 param-name=networks param-key=0/ip
Error: Key 0/ip not found in map
Jul 25 10:50:10 DEBUG [10.1.0.74] xe vm-param-get uuid=7901c950-2345-bcba-a524-796e9fcbf968 param-name=networks param-key=0/ip
Error: Key 0/ip not found in map
Jul 25 10:50:12 DEBUG [10.1.0.74] xe vm-param-get uuid=7901c950-2345-bcba-a524-796e9fcbf968 param-name=networks param-key=0/ip
```

Without the specific logger i think the date time and log level will be logged as well etc... making it less readable also it is a `logging.StreamHandler()` it means it can print line by line if the calls stream to stdout until it returns - i'm not sure the default logging does that.

Reply:
OTOH (regardless of the issue that it does not get redirected together with the debug output, which would have to be dealt with) I found that not having any prefix is a bit confusing, and using `>` and `<` for output and input in logs is not unusal.  Also the timestamping could be still be useful for long-running commands.

Reply:
From the `--log-file` output (which would benefit from getting timestamps in its formatter), we now get things like:
```
INFO     root:common.py:81 Wait for VMs running
DEBUG    root:commands.py:100 [10.1.9.124] xe vm-param-get uuid=503edd57-6f9b-b503-e33d-41e9cb638c7b param-name=power-state
DEBUG    root:commands.py:108 > running
DEBUG    root:commands.py:100 [10.1.9.124] xe vm-param-get uuid=c1886f47-b251-05af-f630-b8d795fc6c8e param-name=power-state
DEBUG    root:commands.py:108 > running
INFO     root:common.py:81 Wait for VM IP
DEBUG    root:commands.py:100 [10.1.9.124] xe vm-param-get uuid=503edd57-6f9b-b503-e33d-41e9cb638c7b param-name=networks param-key=0/ip
DEBUG    root:commands.py:108 > Error: Key 0/ip not found in map
DEBUG    root:commands.py:100 [10.1.9.124] xe vm-param-get uuid=503edd57-6f9b-b503-e33d-41e9cb638c7b param-name=networks param-key=0/ip
DEBUG    root:commands.py:108 > Error: Key 0/ip not found in map
```

Reply:
Also we could want to get a look at https://docs.pytest.org/en/8.2.x/how-to/logging.html#caplog-fixture

Reply:
This is a big topic and we went through many iterations on this, so let's extract it to a separate PR.